src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -80,35 +111,52 @@ public void Clear()
             _cancellationTokenSource.Cancel();
       }
 
-      public SimulationResults[] RunConcurrently()
+      public ConcurrentSimulationResults[] RunConcurrently()
       {
          //Currently we only allow for running simulations or simulation batches, but not both
          if (_simulationBatches.Count > 0 && _simulations.Count > 0)
-            //Temporal Exception. We should allow for mixing both use cases but we need to discuss first
             throw new OSPSuiteException(Error.InvalidMixOfSimulationAndSimulationBatch);
 
          _cancellationTokenSource = new CancellationTokenSource();
          if (_simulations.Count > 0)
          { 
-            var results = _concurrentManager.RunAsync(
+            return _concurrentManager.RunAsync(
                NumberOfCores,
                _cancellationTokenSource.Token,
                _simulations,
                runSimulation
-            ).Result;
-            return _simulations.Select(s => results[s]).ToArray();
+            ).Result.Values.ToArray();
          }
 
          if (_simulationBatches.Count > 0)
-            return null;
+         {
+            return _concurrentManager.RunAsync<SimulationBatchRunOptions, ConcurrentSimulationResults>(
```

Comment:
remove all those generic types They can be inferred by the compiler

Reply:
Done